site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx

```diff
@@ -89,6 +90,56 @@ const getMode = (vehicleMode: Modes | VehicleMode) => {
     return mode;
 };
 
+const groupByJourneyPattern = (routesData: RouteWithServiceInfoPreformatted[]): RouteWithServiceInfo[] => {
```

Comment:
There's quite a bit going on in this function, I would suggest brining this out into the utils file and making a unit test to check it correctly formats it as we expect. And if we know of any edge cases to test how this function would handle those edge cases.

Reply:
done